### PR TITLE
alt-svc frame origin checks from rfc

### PR DESCRIPTION
**Pull request by HoneyryderChuck**
Added the origin checks from the RFC for the `alt-svc` frame.

### CODE REVIEW

File: lib/http/2/connection.rb [hunk 1]
```diff
@@ -450,7 +450,14 @@ def connection_management(frame)
           @state = :closed
           @closed_since = Time.now
           emit(:goaway, frame[:last_stream], frame[:error], frame[:payload])
-        when :altsvc, :blocked
+        when :altsvc
+          # 4.  The ALTSVC HTTP/2 Frame
+          # An ALTSVC frame on stream 0 with empty (length 0) "Origin"
+          # information is invalid and MUST be ignored.
+          if frame[:origin] && !frame[:origin].empty?
```
Comment (igrigorik) [inline lib/http/2/connection.rb]: For stream you have `.nil?`check, any reason why you omitted it here?

Comment (HoneyryderChuck) [reply lib/http/2/connection.rb]: checking that the frame exists and is not empty was the condition, there was no gain in being explicit with `!frame[:origin].nil?`, IMO.

Comment (igrigorik) [reply lib/http/2/connection.rb]: So why do check for nil in stream.rb? Should we remove it there too?

Comment (HoneyryderChuck) [reply lib/http/2/connection.rb]: I see what you mean. I'm out til Monday, I'll sort it out by then.

Comment (HoneyryderChuck) [reply lib/http/2/connection.rb]: done

**Comment by igrigorik** [comment]
Nope, sorry missed your update earlier. Merging. Thanks!

**Pull request merged**